modules/sharedaddy/recaptcha.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * Class that handles reCAPTCHA.
+ */
+class Jetpack_ReCaptcha {
+
+	/**
+	 * URL to which requests are POSTed.
+	 *
+	 * @const string
+	 */
+	const VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify';
+
+	/**
+	 * Site key to use in HTML code.
+	 *
+	 * @var string
+	 */
+	private $site_key;
+
+	/**
+	 * Shared secret for the site.
+	 *
+	 * @var string
+	 */
+	private $secret_key;
+
+	/**
+	 * Config for reCAPTCHA instance.
+	 *
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * Error codes returned from reCAPTCHA API.
+	 *
+	 * @see https://developers.google.com/recaptcha/docs/verify
+	 *
+	 * @var array
+	 */
+	private $error_codes = array(
+		'missing-input-secret'   => 'The secret parameter is missing',
+		'invalid-input-secret'   => 'The secret parameter is invalid or malformed',
+		'missing-input-response' => 'The response parameter is missing',
+		'invalid-input-response' => 'The response parameter is invalid or malformed',
+	);
+
+	/**
+	 * Create a configured instance to use the reCAPTCHA service.
+	 *
+	 * @param string $site_key   Site key to use in HTML code.
+	 * @param string $secret_key Shared secret between site and reCAPTCHA server.
+	 * @param array  $config     Config array to optionally configure reCAPTCHA instance.
+	 */
+	public function __construct( $site_key, $secret_key, $config = array() ) {
+		$this->site_key   = $site_key;
+		$this->secret_key = $secret_key;
+		$this->config     = wp_parse_args( $config, $this->get_default_config() );
+	}
+
+	/**
+	 * Get default config for this reCAPTCHA instance.
+	 *
+	 * @return array Default config
+	 */
+	public function get_default_config() {
+		return array(
+			'language'       => get_locale(),
+			'tag_class'      => 'g-recaptcha',
+			'tag_attributes' => array(
+				'theme'    => 'light',
+				'type'     => 'image',
+				'tabindex' => 0,
+			)
+		);
+	}
+
+	/**
+	 * Calls the reCAPTCHA siteverify API to verify whether the user passes
+	 * CAPTCHA test.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-respones' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return bool|WP_Error Returns true if verified. Otherwise WP_Error is returned.
+	 */
+	public function verify( $response, $remote_ip ) {
+		// No need make a request if response is empty.
+		if ( empty( $response ) ) {
+			return new WP_Error( 'missing-input-response', $this->error_codes['missing-input-response'], 400 );
+		}
+
+		$resp = wp_remote_post( self::VERIFY_URL, $this->get_verify_request_params( $response, $remote_ip ) );
+		if ( is_wp_error( $resp ) ) {
+			return $resp;
+		}
+
+		$resp_decoded = json_decode( wp_remote_retrieve_body( $resp ), true );
+		if ( ! $resp_decoded ) {
+			return new WP_Error( 'invalid-json', 'Invalid JSON', 400 );
+		}
+
+		if ( isset( $resp_decoded['success'] ) && true === $resp_decoded['success'] ) {
+			return true;
```

Comment:
It's a bit weird to have the success case in the middle of the class. It makes readability of the method much harder. Can/should we move it to the end?